### PR TITLE
6 more DMC + 2 Tokenscripts

### PR DESCRIPTION
**Pull request by Simisays**
Mana Cannons
Rasputin, the Oneiromancer
The Reaver Cleaver
Two-Headed Hellkite
Xira, the Golden Sting
Zeriam, Golden Wind

### CODE REVIEW

File: forge-gui/res/tokenscripts/w_2_2_knight_protectionred.txt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+Name:Knight Token
```
Comment (paulsnoops) [inline forge-gui/res/tokenscripts/w_2_2_knight_protectionred.txt:1]: please name the file w_2_2_knight_pro_red

File: forge-gui/res/cardsfolder/upcoming/rasputin_the_neiromancer.txt [hunk 2]
```diff
@@ -0,0 +1,12 @@
+Name:Rasputin, the Oneiromancer
+ManaCost:1 W U
+Types:Legendary Creature Human Wizard
+PT:4/1
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigCounter | TriggerDescription$ When CARDNAME enters the battlefield, put a dream counter on it for each opponent you have. Each opponent creates a 1/1 red Goblin creature token.
+SVar:TrigCounter:DB$ PutCounter | CounterType$ Dream | CounterNum$ X | SubAbility$ DBToken
+SVar:DBToken:DB$ Token | TokenAmount$ 1 | TokenScript$ r_1_1_goblin | TokenOwner$ Opponent 
+A:AB$ Mana | Cost$ T SubCounter<X/DREAM> | Produced$ C | Amount$ Y |  SpellDescription$ Add that much {C}.
+A:AB$ Token | Cost$ T SubCounter<1/DREAM> | TokenScript$ w_2_2_knight_protectionred | SpellDescription$ Create a 2/2 white Knight creature token with protection from red.
```
Comment (paulsnoops) [inline forge-gui/res/cardsfolder/upcoming/rasputin_the_neiromancer.txt:9]: TokenScript$ w_2_2_knight_pro_red

File: forge-gui/res/tokenscripts/w_2_2_knight_protectionred.txt (hunk 1 above)
Comment (Simisays) [reply forge-gui/res/tokenscripts/w_2_2_knight_protectionred.txt]: Done !

File: forge-gui/res/cardsfolder/upcoming/two_headed_hellkite.txt [hunk 3]
```diff
@@ -0,0 +1,10 @@
+Name:Two-Headed Hellkite
```
Comment (paulsnoops) [inline forge-gui/res/cardsfolder/upcoming/two_headed_hellkite.txt:1]: Just FYI.. it's normally fine to leave "two-headed" in the file name when the card name has the hyphen present. We only really use _ for a space. But it's up to you, not a big deal. Both ways are fine here.

Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/two_headed_hellkite.txt:1]: Ok will keep in mind !

File: forge-gui/res/cardsfolder/upcoming/mana_cannons.txt [hunk 4]
```diff
@@ -0,0 +1,8 @@
+Name:Mana Cannons
+ManaCost:2 R
+Types:Enchantment
+T:Mode$ SpellCast | ValidCard$ Card.MultiColor | ValidActivatingPlayer$ You | Execute$ TrigDamage | TriggerZones$ Battlefield | TriggerDescription$ Whenever you cast a multicolored spell, CARDNAME deals X damage to any target, where X is the number of colors that spell is.
+SVar:TrigDamage:DB$ DealDamage | ValidTgts$ Creature,Player,Planeswalker | TgtPrompt$ Select any target | NumDmg$ X
+SVar:X:TriggeredCard$CardNumColors
+SVar:BuffedBy:Card.MultiColor
+Oracle:Whenever you cast a multicolored spell, Mana Cannons deals X damage to any target, where X is the number of colors that spell is.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/mana_cannons.txt:8]: AI:RemoveDeck:Random - the AI shouldn't put this in a mono red deck... but we don't have a more specific way to tell it that yet™️

File: forge-gui/res/cardsfolder/upcoming/rasputin_the_oneiromancer.txt [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+Types:Legendary Creature Human Wizard
+PT:4/1
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigCounter | TriggerDescription$ When CARDNAME enters the battlefield, put a dream counter on it for each opponent you have. Each opponent creates a 1/1 red Goblin creature token.
+SVar:TrigCounter:DB$ PutCounter | CounterType$ Dream | CounterNum$ X | SubAbility$ DBToken
+SVar:DBToken:DB$ Token | TokenAmount$ 1 | TokenScript$ r_1_1_goblin | TokenOwner$ Opponent 
+A:AB$ Mana | Cost$ T SubCounter<Y/DREAM> | Produced$ C | Amount$ Y |  SpellDescription$ Add that much {C}.
+A:AB$ Token | Cost$ T SubCounter<1/DREAM> | TokenScript$ w_2_2_knight_pro_red | SpellDescription$ Create a 2/2 white Knight creature token with protection from red.
+SVar:X:PlayerCountOpponents$Amount
+SVar:Y:Count$xPaid
+Oracle:When Rasputin, the Oneiromancer enters the battlefield, put a dream counter on it for each opponent you have. Each opponent creates a 1/1 red Goblin creature token.\n{T}: Remove one or more dream counters from Raputin: Add that much {C}.\n{T}: Remove a dream counter from Rasputin: Create a 2/2 white Knight creature token with protection from red.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/rasputin_the_oneiromancer.txt:12]: `Raputin`

File: forge-gui/res/cardsfolder/upcoming/rasputin_the_oneiromancer.txt [hunk 6]
```diff
@@ -0,0 +1,12 @@
+Name:Rasputin, the Oneiromancer
+ManaCost:1 W U
+Types:Legendary Creature Human Wizard
+PT:4/1
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigCounter | TriggerDescription$ When CARDNAME enters the battlefield, put a dream counter on it for each opponent you have. Each opponent creates a 1/1 red Goblin creature token.
+SVar:TrigCounter:DB$ PutCounter | CounterType$ Dream | CounterNum$ X | SubAbility$ DBToken
+SVar:DBToken:DB$ Token | TokenAmount$ 1 | TokenScript$ r_1_1_goblin | TokenOwner$ Opponent 
+A:AB$ Mana | Cost$ T SubCounter<Y/DREAM> | Produced$ C | Amount$ Y |  SpellDescription$ Add that much {C}.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/rasputin_the_oneiromancer.txt:8]: ` |..Spell` > ` |.Spell`

File: forge-gui/res/cardsfolder/upcoming/mana_cannons.txt (hunk 4 above)
Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/mana_cannons.txt:8]: > yet™️

Ooh the foreshadowing  ;)

File: forge-gui/res/cardsfolder/upcoming/xira_the_golden_sting.txt [hunk 7]
```diff
@@ -0,0 +1,12 @@
+Name:Xira, the Golden Sting
+ManaCost:1 B R G
+Types:Legendary Creature Insect Assassin 
+PT:3/3
+K:Flying
+K:Haste
+T:Mode$ Attacks | ValidCard$ Card.Self | Execute$ TrigEggCounter | TriggerDescription$ Whenever CARDNAME attacks, put an egg counter on another target creature without an egg counter on it. When that creature dies, if it has an egg counter on it, draw a card and create a 1/1 black Insect creature token with flying.
+SVar:TrigEggCounter:DB$ PutCounter | CounterType$ EGG | CounterNum$ 1 | ValidTgts$ Creature.Other+counters_EQ0_EGG | RememberTargets$ True | TgtPrompt$ Put an egg counter on another target creature without an egg counter on it. | SubAbility$ DelayedTrigger
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/xira_the_golden_sting.txt:8]: RememberTargets not needed

File: forge-gui/res/cardsfolder/upcoming/the_reaver_cleaver.txt [hunk 8]
```diff
@@ -0,0 +1,10 @@
+Name:The Reaver Cleaver
+ManaCost:2 R
+Types:Legendary Artifact Equipment
+K:Equip:3
+S:Mode$ Continuous | Affected$ Creature.EquippedBy | AddPower$ 1 | AddToughness$ 1 | AddKeyword$ Trample | Description$ Equipped creature gets +1/+1 and has trample.
+T:Mode$ DamageDone | ValidSource$ Creature.EquippedBy | ValidTarget$ Player | CombatDamage$ True | Execute$ TrigToken | TriggerZones$ Battlefield | TriggerDescription$ Whenever this creature deals combat damage to a player or planeswalker, create that many Treasure tokens.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/the_reaver_cleaver.txt:6]: ValidTarget missing PW

File: forge-gui/res/cardsfolder/upcoming/the_reaver_cleaver.txt [hunk 9]
```diff
@@ -3,7 +3,7 @@ ManaCost:2 R
 Types:Legendary Artifact Equipment
 K:Equip:3
 S:Mode$ Continuous | Affected$ Creature.EquippedBy | AddPower$ 1 | AddToughness$ 1 | AddKeyword$ Trample | Description$ Equipped creature gets +1/+1 and has trample.
-T:Mode$ DamageDone | ValidSource$ Creature.EquippedBy | ValidTarget$ Player | CombatDamage$ True | Execute$ TrigToken | TriggerZones$ Battlefield | TriggerDescription$ Whenever this creature deals combat damage to a player or planeswalker, create that many Treasure tokens.
+T:Mode$ DamageDone | ValidSource$ Creature.EquippedBy | ValidTarget$ Player,Planeswalker.YouCtrl | CombatDamage$ True | Execute$ TrigToken | TriggerZones$ Battlefield | TriggerDescription$ Whenever this creature deals combat damage to a player or planeswalker, create that many Treasure tokens.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/the_reaver_cleaver.txt:6]: The YouCtrl doesn't make sense

File: forge-gui/res/cardsfolder/upcoming/zeriam_golden_wind.txt [hunk 10]
```diff
@@ -0,0 +1,8 @@
+Name:Zeriam, Golden Wind
+ManaCost:3 W
+Types:Legendary Creature Griffin
+PT:3/4
+K:Flying
+T:Mode$ DamageDone | ValidSource$ Griffin.YouCtrl | ValidTarget$ Opponent | CombatDamage$ True | TriggerZones$ Battlefield | Execute$ TrigToken | TriggerDescription$ Whenever a Griffin you control deals combat damage to a player, create a 2/2 white Griffin creature token with flying.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/zeriam_golden_wind.txt:6]: `ValidTarget$ Player`

always try and reproduce the wording as close as possible for all corner cases
here the combat damage could be redirected back to you and you should still get the trigger

**Review by tool4ever — APPROVED** [review]
Nice work!

**Pull request merged**